specs/lightclient/beacon-chain.md

```diff
@@ -561,19 +581,106 @@ def process_justification_and_finalization(state: BeaconState) -> None:
 
 #### New `process_rewards_and_penalties`
 
-*Note*: The function `process_rewards_and_penalties` is modified to use participation flag deltas.
+*Note*: The function `process_rewards_and_penalties` is modified to use participation flag deltas, and is broken up into the rewards and penalties-focused parts, which now work very differently.
 
 ```python
 def process_rewards_and_penalties(state: BeaconState) -> None:
+    process_rewards(state)
+    process_penalties(state)
+```
+
+##### `process_rewards`
+
+```python
+def process_rewards(state: BeaconState) -> None:
     # No rewards are applied at the end of `GENESIS_EPOCH` because rewards are for work done in the previous epoch
     if get_current_epoch(state) == GENESIS_EPOCH:
         return
-    flag_deltas = [get_flag_deltas(state, flag, numerator) for (flag, numerator) in get_flags_and_numerators()]
-    deltas = flag_deltas + [get_inactivity_penalty_deltas(state)]
-    for (rewards, penalties) in deltas:
+    flag_rewards = [get_flag_rewards(state, flag, numerator) for (flag, numerator) in get_flags_and_numerators()]
+    for rewards in flag_rewards:
         for index in range(len(state.validators)):
             increase_balance(state, ValidatorIndex(index), rewards[index])
-            decrease_balance(state, ValidatorIndex(index), penalties[index])
+```
+
+##### `process_penalties`
+
+```python
+def process_penalties(state: BeaconState) -> None:
+    # A validator's leak_score updates as follows:
+    # (i)  If there was an inactivity leak in a given epoch, anyone who participated in that epoch
+    #      has their score decrease by 1, anyone who did not has their score increase by LEAK_SCORE_BIAS
+    # (ii) If there was not an inactivity leak in a given epoch, anyone who participated in that epoch
+    #      has their score decrease by 1
+    # The code below implements this, though amortizing the "increase if absent" component until the end
+    # of the ACTIVATION_EXIT_PERIOD
+    
+    if is_in_inactivity_leak(state):
+        leak_score_decrease = LEAK_SCORE_BIAS + 1
```

Comment:
I don't understand the rationale to `+1` here. Perhaps some code comments to clarify would be nice : )